bench-tps/src/bench.rs

```diff
@@ -402,29 +402,33 @@ fn poll_blockhash<T: Client>(
     client: &Arc<T>,
     id: &Pubkey,
 ) {
-    let mut blockhash_time;
+    let mut blockhash_last_updated = Instant::now();
     loop {
-        blockhash_time = Instant::now();
-        loop {
+        let blockhash_updated = {
             let old_blockhash = *blockhash.read().unwrap();
             if let Ok((new_blockhash, _fee)) = client.get_new_blockhash(&old_blockhash) {
                 *blockhash.write().unwrap() = new_blockhash;
-                break;
+                blockhash_last_updated = Instant::now();
+                true
             } else {
-                if blockhash_time.elapsed().as_secs() > 30 {
-                    panic!("Blockhash is not updating");
+                if blockhash_last_updated.elapsed().as_secs() > 30 {
+                    error!("Blockhash is not updating");
                 }
-                sleep(Duration::from_millis(50));
-                continue;
+                false
             }
-        }
+        };
 
-        let balance = client.get_balance(id).unwrap_or(0);
-        metrics_submit_lamport_balance(balance);
+        if blockhash_updated {
+            let balance = client.get_balance(id).unwrap_or(0);
+            metrics_submit_lamport_balance(balance);
+        }
 
         if exit_signal.load(Ordering::Relaxed) {
             break;
         }
+
+        // Wait a slot before checking again
+        sleep(Duration::from_millis(400));
```

Comment:
I disagree that we should be less aggressive at polling. This could cause the blockhash to be 800ms older than it needs to be.

Reply:
@sakridge is it bad to have an 800ms old blockhash? Blockhashes are active for about 2min right?

Reply:
Yes, there can be significant latency in the system from forwarding/tx processing so you want to have the absolute newest blockhash to prevent blockhash too old errors which lower throughput. This is a benchmark designed to push the system to the max, so nodes can and will be behind with the TX load.

Reply:
Ah got it, thanks for the explanation, will switch back